src/zookeeperLock.ts

```diff
@@ -452,8 +600,28 @@ export class ZookeeperLock extends EventEmitter {
     };
 
 
-    private checkBail = () => {
-        return this.timedOut || this.destroyed || (this.client && (<any>this.client.getState()).name !== 'SYNC_CONNECTED');
+    /**
+     * are we on the happy path to continue locking?
+     * @returns {boolean|zk.Client}
+     */
+    private continueLocking = () => {
+        return this.state === ZookeeperLock.States.LOCKING && (this.client && (<any>this.client.getState()).name === 'SYNC_CONNECTED');
```

Comment:
can the state name be `CONNECTED` here? whats the difference between SYNC_CONNECTED and CONNECTED